### PR TITLE
ci: remove redundant conditional

### PR DESCRIPTION
**Pull request by Fdawgs**
#### Checklist

- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -44,6 +44,5 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: fastify/github-action-merge-dependabot@v2.0.0
-        if: ${{ github.actor == 'dependabot[bot]' && github.event_name == 'pull_request' }}
```
Comment (aboutlo) [inline .github/workflows/ci.yml:47]: hey @Fdawgs If I remember correctly we added this because we were not running the checks on the pull_request.

Comment (mcollina) [reply .github/workflows/ci.yml:47]: it's not needde anymore

**Review by mcollina — APPROVED** [review]
lgtm

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Fdawgs) [reply .github/workflows/ci.yml:47]: Hi @aboutlo, as @mcollina said, with v2.0.0 of the action it handles all this for us. So the 'if' conditional is no longer needed.

**Pull request merged**